src/npc_class.h

```diff
@@ -33,6 +33,7 @@ class distribution
 
         distribution operator+( const distribution &other ) const;
         distribution operator*( const distribution &other ) const;
+        distribution& operator=( const distribution &other );
```

Comment:
Here is the whole story: [https://github.com/CleverRaven/Cataclysm-DDA/issues/17714](https://github.com/CleverRaven/Cataclysm-DDA/issues/17714). I suspect clang compiler doesn't copy the `std::function` properly with optimization switch on, causes the crash.